node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/CapacityPolicies.java

```diff
@@ -174,7 +174,7 @@ private NodeResources smallestSharedResources() {
     public ClusterSpec decideExclusivity(Capacity capacity, ClusterSpec requestedCluster) {
         if (capacity.cloudAccount().isPresent()) return requestedCluster.withExclusivity(true); // Implicit exclusive
         boolean exclusive = requestedCluster.isExclusive() && (capacity.isRequired() || zone.environment() == Environment.prod);
-        return requestedCluster.withExclusivity(exclusive);
+        return requestedCluster.withExclusivity(exclusive).withProvisionForApplication(exclusive);
```

Comment:
There are other uses of the `Builder.exclusive(...)` as well. Did you check those?

Reply:
💯